### DISCUSSION TITLE
HashExp With Relationships

### DISCUSSION THREAD

**Discussion by hmajid2301** [pocketbase/pocketbase#1644] [Q&A]
How can I use a HashExp to query for a column which is a relationship.

For example I have a table which looks like:

![image](https://user-images.githubusercontent.com/998807/213581109-01b1147e-13c3-4e6b-94a1-2dbeed059704.png)
![image](https://user-images.githubusercontent.com/998807/213581454-8f029faa-aefb-4e97-b10d-c870a73b2dea.png)

Where `user` has a foreign key relationship with `users` tables.

```golang
func GetByUserID(name string, userID string) (*models.Record, error) {
	expr1 := dbx.HashExp{"user": userID}
	records, err := s.client.FindRecordsByExpr(collectionName, expr1)
       // returns 0 records even though there are records related to this UserId
}
````

How can do the query correctly using the userID ? 

Thanks!

**Comment by ganigeorgiev** [comment]
Hm, if `user` is a single `relation` field (aka. "Max select = 1") then the above should work. If it is a multi-`relation` field (aka. "Max select != 1"), then you'll have to use [`json_extract`](https://www.sqlite.org/json1.html#jex) since multi-value fields are stored as json array.

Make sure that `collectionName` and `userID` are correct.
You can also inspect the generated query in the console by starting PocketBase with `--debug`.
